### PR TITLE
Dataset home improvements and icons for sidebar

### LINKED ISSUE #3031: Data / Dataset home improvements and icons for sidebar
https://www.figma.com/proto/IFqcY4iCKC9ukJkgfSRKEA/1910-Gobierto-Datos-Abiertos-Bocetos?node-id=594%3A375&viewport=-1135%2C884%2C0.5&scaling=min-zoom

I propose that for the time being (until we have the full scrutinizer) we a block with the list of columns. In the future it may also make sense to add a description for each column. 

Tasks would be: 

- sidebar: icon for column type instead of the current plain text. Thoughts?
- dataset: block with the list of columns in the dataset (I'm not too fond of showing them on three columns, but it we just put one, we get lots of empty space. In the future if we develop the dictionary or scrutinizer, we would iterate this)
- dataset: also let's add a button "Preview" besides the download
- dataset: show heading for blocks. Now we are showing the Queries block, but without its header
- dataset: Don't show a block if it doesn't have content

### PR DESCRIPTION
**Pull request by jorgeatgu**
Closes #3031

## :v: What does this PR do?

- sidebar: Include icons for columns type instead of the current plain text
- Summary tab: Descritption is a new block with the list of columns of the Dataset 
- Summary tab: New button to preview 
- Summary tab: include headings for blocks and hide blocks and labels without content.
- Queries modal: Fixed a bug in modal queries component, if you clicked
 inside the modal goes to closed.

## :mag: How should this be manually tested?

Staging

### Before this PR

3031_before 2020-05-14 10_37_07

### After this PR

3031 after 2020-05-14 at 10 58 07 2020-05-14 11_00_19

### CODE REVIEW

File: app/assets/stylesheets/module-data.scss [hunk 1]
```diff
@@ -319,10 +320,18 @@
...
+    font-family: $font_text;
+    font-weight: 600;
     min-width: 250px;
-    margin: 1rem 0;
+    margin: 1rem 1rem .5rem .25rem;
     position: relative;
+    text-transform: unset;
+    vertical-align: top;
+
+    i {
```
Comment (Crashillo) [inline app/assets/stylesheets/module-data.scss]: tagging `i`, use `important`. I hope you got a good excuse, little boy. 😄

File: app/assets/stylesheets/module-data.scss [hunk 2]
```diff
@@ -1087,4 +1098,53 @@
...
+  &-summary-header-btns {
+    margin-top: 1.5rem;
+  }
+
+  &-columns-icon-boolean::before,
+  &-columns-icon-decimal::before,
+  &-columns-icon-hstore::before,
+  &-columns-icon-integer::before,
+  &-columns-icon-inet::before {
+    content: "\f292";
```
Comment (Crashillo) [inline app/assets/stylesheets/module-data.scss:1133]: What is that? icons?

File: app/javascript/gobierto_data/webapp/components/commons/Description.vue [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            class="gobierto-data-columns-icon-description"
+          >
+            <i
+              :class="'fas gobierto-data-columns-icon gobierto-data-columns-icon-' + Object.values(element)[0]"
+              style="color: var(--color-base);"
+            />
+            <span class="gobierto-data-sidebar-datasets-links-columns-text">
+              {{ Object.keys(element)[0] }}
+            </span>
+          </div>
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Description.vue]: Casi cuela

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue [hunk 4]
```diff
@@ -2,51 +2,50 @@
...
+            {{ labelSubject }}
+          </span>
+          <a
+            href="#"
+            class="gobierto-data-summary-header-container-text-link"
+          >
+            {{ categoryDataset }}
+          </a>
+        </div>
+      </template>
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Info.vue]: You see you're using almost the same markup, right? Do you know how do we call stuff like that, right? Yeah, component

Comment (Crashillo) [reply app/javascript/gobierto_data/webapp/components/commons/Info.vue]: 3rd time rule: "if you need to copy paste the same thing 3 times, you're doing something wrong"

File: app/javascript/gobierto_data/webapp/components/commons/Queries.vue [hunk 5]
```diff
@@ -144,6 +141,9 @@ export default {
     };
 
     this.editor = CodeMirror.fromTextArea(this.$refs.querySnippet, cmOption);
+
+    this.showPrivateQueries = this.privateQueries.length ? false : true
+    this.showPublicQueries = this.publicQueries.length ? false : true
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Queries.vue]: What about this?
```suggestion
    this.showPrivateQueries = !!this.privateQueries.length
    this.showPublicQueries = !!this.publicQueries.length
```

File: app/javascript/gobierto_data/webapp/components/sets/SummaryTab.vue [hunk 6]
```diff
@@ -5,36 +5,79 @@
...
+          {{ labelQueries }}
+        </h2>
+      </template>
+      <div>
+        <Queries
+          v-if="showYourQueries"
+          :private-queries="privateQueries"
+          :public-queries="publicQueries"
+        />
+      </div>
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/sets/SummaryTab.vue]: Perhaps you don't need the extra `div`

File: app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue [hunk 7]
```diff
@@ -36,25 +36,31 @@
...
-        v-clickoutside="closeQueriesModal"
-        :class="{ 'remove-label' : removeLabelBtn }"
-        :title="labelButtonQueries"
-        class="btn-sql-editor"
-        icon="list"
-        color="var(--color-base)"
-        background="#fff"
-        @click.native="isQueriesModalActive = !isQueriesModalActive"
-      />
+      <!-- //Closable directive doesn't work with component, so i've used the html attribute :\ -->
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue:37]: Events doesn't work on Component unless you set `.native`, have you tried?

**Review by Crashillo — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/module-data.scss (hunk 2 above)
Comment (jorgeatgu) [reply app/assets/stylesheets/module-data.scss:1133]: Yes, icons for sidebar

File: app/javascript/gobierto_data/webapp/components/commons/Queries.vue (hunk 5 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/commons/Queries.vue]: Much better!

File: app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue (hunk 7 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorHeader.vue:37]: Yes, I tried, the problem isn't with the component, it's with the closable directive

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue (hunk 4 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/commons/Info.vue]: you're DRY

File: app/assets/stylesheets/module-data.scss [hunk 8]
```diff
@@ -1087,4 +1094,53 @@
     background-color: transparent;
   }
 
+  &-description {
+    height: auto;
+    -webkit-columns: 3;
+    -moz-columns: 3;
+    columns: 3;
+    -moz-column-fill: balance;
```
Comment (Crashillo) [inline app/assets/stylesheets/module-data.scss]: Unnecessary

https://caniuse.com/#search=columns

File: app/javascript/gobierto_data/webapp/components/commons/Description.vue [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+          <Caret :rotate="showDescription" />
+          {{ labelDescription }}
+        </h2>
+      </template>
+      <div class="gobierto-data-description">
+        <template
+          v-for="(row, i) in numberOfRows"
+        >
+          <div
+            v-for="{ column, type } in arrayOfObjectsColumns.slice(i * itemsPerRow, (i + 1) * itemsPerRow)"
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Description.vue]: What did this do?

File: app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue [hunk 10]
```diff
@@ -5,6 +5,7 @@
       :text="labelDownloadData"
       icon="download"
       color="var(--color-base)"
+      icon-color="rgba(var(--color-base-string), .5)"
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue:7]: Just one question: if the common button color (both) is the `base`, wouldn't be better to set as default fallback in the `Button` component? In that way, whenever we wanna use it and do not need an additional color, we don't need to pass such props

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue [hunk 11]
```diff
@@ -1,52 +1,30 @@
...
+        icon="calendar"
+        icon-color="var(--color-base)"
+        opacity=".25"
+        :label="labelFrequency"
+        :text="frequencyDataset"
+      />
+      <InfoBlockText
+        v-if="categoryDataset"
+        icon="tag"
+        icon-color="var(--color-base)"
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Info.vue]: Look at the previous comment about the fallback color, I think these are a good examples

**Review by Crashillo — CHANGES_REQUESTED** [review]

File: app/javascript/gobierto_data/webapp/components/commons/Description.vue (hunk 9 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/commons/Description.vue]: At first it made sense, now that we build the columns with CSS it doesn't make sense. Refactor

File: app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue (hunk 10 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue:7]: Yes. Good point, as usual. So:

```
color: {
      type: String,
      default: 'var(--color-base)'
    }

iconColor: {
      type: String,
      default: 'var(--color-base)'
    },
```

File: config/locales/gobierto_data/views/ca.yml [hunk 12]
```diff
@@ -39,6 +40,7 @@ ca:
       guide: Guia per utilitzar l'editor SQL
       modifiedQuery: Consulta modificada
       modifiedVisualization: Visualització modificada
+      preview: Preview
```
Comment (Crashillo) [inline config/locales/gobierto_data/views/ca.yml]: Previsualitzar (if it's a button or link, better Previsualitzeu)

File: config/locales/gobierto_data/views/es.yml [hunk 13]
```diff
@@ -39,6 +40,7 @@ es:
       guide: Guía para usar el editor SQL
       modifiedQuery: Consulta modificada
       modifiedVisualization: Visualización modificada
+      preview: Previsualitzar
```
Comment (Crashillo) [inline config/locales/gobierto_data/views/es.yml]: Previsualizar

**Review by Crashillo — CHANGES_REQUESTED** [review]

**Review by ferblape — CHANGES_REQUESTED** [review]
The queries tab seems broken

Screenshot 2020-05-20 at 07 19 35

**Comment by jorgeatgu** [comment]
> The queries tab seems broken
> 
> Screenshot 2020-05-20 at 07 19 35

Yes and no. If the dataset doesn't contain queries(public or privates) this block doesn't appear. In this case, it looks a bit strange, but if we show the block only appears

> Tus consultas(0)

Maybe we should show a string as "No hay consultas disponibles"
Screenshot 2020-05-20 at 11 24 24


@furilo

**Comment by furilo** [comment]
The block should not appear only in the home.

**Review by Crashillo — APPROVED** [review]

**Comment by ferblape** [comment]
But this is not consistent:

Screenshot 2020-05-20 at 17 06 48

Screenshot 2020-05-20 at 17 06 50

Why not show the message "No hay consultas" like we do in visualizations?

**Comment by furilo** [comment]
OK

**Comment by jorgeatgu** [comment]
> But this is not consistent:
> 
> Screenshot 2020-05-20 at 17 06 48
> 
> Screenshot 2020-05-20 at 17 06 50
> 
> Why not show the message "No hay consultas" like we do in visualizations?

Done!

Screenshot 2020-05-22 at 14 54 23

**Pull request merged**